File: kickstart/2021/RoundH/transform_string.py
import numpy as np
T = int(input())
for t in range(1, T + 1):
    S = input().strip()
    F = input().strip()
    dist = []
    for i, cf in enumerate(F):
        dist.append(
            [min(abs(ord(cf) - ord(cs)), 25 - abs(ord(cf) - ord(cs)) + 1) for cs in S])
    res = 0
    for i in range(len(S)):
        res += min(np.array(dist)[:, i])
    print('Case #{}: {}'.format(t, res))

File: kickstart/2021/RoundH/transform_string.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int T;
    cin >> T;
    string letters = "abcdefghijklmnopqrstuvwxyz";
    int len_letters = static_cast<int>(letters.size());
    for (int t = 1; t <= T; ++t)
    {
        string S, F;
        cin >> S;
        cin >> F;
        vector<vector<int>> dist;
        dist.resize(F.size());
        for (int i = 0; i < F.size(); ++i)
        {
            int af = F[i];
            for (int j = 0; j < S.size(); ++j)
            {
                int as = S[j];
                dist[i].push_back(min(abs(af - as), len_letters - abs(af - as)));
            }
        }
        int res = 0;
        for (int i = 0; i < S.size(); ++i)
        {
            int min_dist = len_letters;
            for (int j = 0; j < F.size(); ++j)
            {
                if (min_dist > dist[j][i])
                {
                    min_dist = dist[j][i];
                }
            }
            res += min_dist;
        }
        cout << "Case #" << t << ": " << res << endl;
    }
}
